### DISCUSSION TITLE
The default export must be a function or server.

### DISCUSSION THREAD

**Discussion by cosbgn** [vercel/vercel#6457] [Help]
Hi,

I can't get any lambdas to work without using the `helpers`. I would like to use the same format as AWS uses so I can easily switch if I need to. I tried the example in the docs: 

```
exports.handler = async function (event, context, callback) {
  return {
    statusCode: 200,
    headers: {},
    body: 'Hello world',
  };
};
```
And :

```
export async function handler(event, context) {
	const users = await db.get_many("users")
	const cr = users.map(u => u.created_at)
	return {
	  statusCode: 200,
	  body: JSON.stringify({ message: cr })
	}
}
```

I always get `The default export must be a function or server.` - However in production it works. The only issue is with `vc dev` - Any idea on how I can get it to work? (p.s. I have the env variables set to handler and helpers=0

**Comment by barelyhuman** [comment]
Could you try this for me and let me know if this solves the issues. 

The reason is because the micro instance requires a default export from the module and the above code exports just a module, this works in prod because there's lamba wrappers around the actual code (not exactly but simplified for explanation).

for the dev it needs micro to understand the above so try adding this to one of them 

```js
export async function handler(event, context) {
	const users = await db.get_many("users")
	const cr = users.map(u => u.created_at)
	return {
	  statusCode: 200,
	  body: JSON.stringify({ message: cr })
	}
}

export default handler;
```

but obviously the dev would still fail since the handler doesn't take the params as per micro so you might wanna translate the same using an interface like 

```js
export async function handler(event, context) {
	const users = await db.get_many("users")
	const cr = users.map(u => u.created_at)
	return {
	  statusCode: 200,
	  body: JSON.stringify({ message: cr })
	}
}

export default (req,res)=>{
/// interface setup 
};
```

**Reply by cosbgn** [reply]
If I understood correctly this is how it should work:
```
export async function handler(event, context) {
	return {
	  statusCode: 200,
	  body: JSON.stringify({ message: "Hey" })
	}
}

export default async (req,res) => {
	const hand = await handler()
	res.json(hand)
};
```
In this case it doesn't work for two reasons:

1 - It still uses the helpers. I would like to avoid the helpers so that I can deploy my code on any other serverless platform if needed.

2 - It doesn't parse the respose, I get the statusCode in the json and the body as a string, not as an object

**Reply by barelyhuman** [reply]
I understand but micro(the underlying executor) for the requests expects you to have the function that way and while I understand your concerns you are still in a way tightly coupling it to AWS lambda, the above will not be usable in case of Google Cloud Run / Actions and you'll need an interface to the actual logic almost always.

**Reply by cosbgn** [reply]
Ok I understand, so this page is is just depriciated: https://vercel.com/docs/runtimes?query=lambda#advanced-usage/advanced-node-js-usage/aws-lambda-api
 
And there isn't really a way to do this. The only way is to use Vercel's micro, right? (At least in dev)

**Reply by barelyhuman** [reply]
No the doc isn't deprecated, but the vercel config for `build.env` is.
Yeah, for dev it's going to be micro , but just to confirm, you `vercel.json` does match the given config? 
also the `builds.env` is now legacy so the support from the team will be limited as well . 
https://vercel.com/docs/configuration#project/build-env

You can try adding the same ENV Variable for NODEJS_AWS_HANDLER_NAME on the actual portal for dev/prod etc and then try vercel dev , theoretically this should work else the way out is the interface helper approach
